sdk/nodejs/output.ts

```diff
@@ -226,13 +242,19 @@ This function may throw in a future version of @pulumi/pulumi.`;
 
                 // Else for *any other* property lookup, succeed the lookup and return a lifted
                 // `apply` on the underlying `Output`.
-                return obj.apply(ob => {
+                return (<any>obj.apply)((ob: any) => {
                     if (ob === undefined || ob === null) {
                         return undefined;
                     }
+                    else if (isUnknown(ob)) {
+                        // If the value of this output is unknown, the result of the access should also be unknown.
+                        // This is conceptually consistent, and also prevents us from returning a "known undefined"
+                        // value from the `ob[prop]` expression below.
+                        return unknown;
+                    }
```

Comment:
ok.  so just to check.  if we had an object with two levels of unknown nesting, i.e. :

```ob ~= { a: unknown, b: { c: unknown, d: 0 } }```

then calling `ob.b` would result in `unknown`, even though `ob.b.d` could technically get you to a known value?

Or is this just saying if *b* is *exactly* the `unknown` value then return `unknown`?

Reply:
> Or is this just saying if b is exactly the unknown value then return unknown?

This one.

Reply:
thanks.